src/compiler/checker.ts

```diff
@@ -11094,24 +11114,29 @@ namespace ts {
         }
 
         function getIndexSymbol(symbol: Symbol): Symbol | undefined {

```

Comment:
You're splitting both `getIndexSymbol` and `getIndexDeclarationOfSymbol` to two functions each, because your code calls them with different inputs, and it's the only place that needs the two new entry points.  One such refactor seems questionable, but two looks like too much.  (But OTOH I can't figure out if there's a way to avoid it.)